compiler/sempass2.nim

```diff
@@ -224,8 +224,25 @@ proc markGcUnsafe(a: PEffects; reason: PNode) =
                                           a.owner, reason.info, {})
 
 when true:
-  template markSideEffect(a: PEffects; reason: typed) =
-    if not a.inEnforcedNoSideEffects: a.hasSideEffect = true
+  template markSideEffect(a: PEffects; reason: typed; useLoc: TLineInfo) =
+    if not a.inEnforcedNoSideEffects:
+      a.hasSideEffect = true
+      if a.owner.kind in routineKinds:
+        var sym: PSym
+        when reason is PNode:
+          if reason.kind == nkSym:
+            sym = reason.sym
+          else:
+            var kind = skUnknown
+            case reason.kind
+            of nkHiddenDeref:
+              kind = skParam
+            else: discard
+            sym = newSym(kind, a.owner.name, nextSymId a.c.idgen,
+                                              a.owner, reason.info, {})
+        else:
+          sym = reason
+        a.owner.sideEffectReasons.add (useLoc, sym)
 else:
   template markSideEffect(a: PEffects; reason: typed) =
```

Comment:
remove this disabled template, and add:
```nim
when false: markGcUnsafe(a, reason)
```
which preserves the orginial intent without introducing code duplication

Reply:
What is the intent of `when false`?
Keep code around that maybe needed later on?

Reply:
yes, and ensures it's still highlighted and parseable, unlike comments